src/main/resources/liquibase/crop_changelog/23_1_0.xml

```diff
@@ -64,5 +64,199 @@
+        </comment>
+        <sql dbms="mysql" splitStatements="true">
+            -- Create a temporary table for names with duplicates
+            CREATE temporary TABLE name_with_duplicates
+                SELECT name.nid, name.gid, name.ntype, nameType.fcode, nameType.ftype, nameType.ftable
+                    FROM names name
+                    INNER JOIN udflds nameType ON nameType.fldno = name.ntype
+                    GROUP  BY name.gid, name.ntype
+                    HAVING Count(*) > 1;
+
+            -- Assign duplicate name with incremented number
+            CREATE temporary TABLE name_with_incremented_number
+            SELECT nid, gid, ntype,
+                CASE WHEN nameincrement = 0
+                    THEN fcode
+                ELSE Concat(fcode, '_', nameincrement)
+                    END AS fcode_with_increment
+                FROM (SELECT names.nid, names.gid, names.ntype, name_with_duplicates.fcode, name_with_duplicates.ftype,
+                    CASE WHEN @previousgid = names.gid  AND @previousntype = names.ntype THEN
+                        @runninggroup := @runninggroup + 1
+                    ELSE @runninggroup := 0
+                        end AS nameIncrement, @previousgid := names.gid, @previousntype := names.ntype
+                    FROM names
+                    INNER JOIN name_with_duplicates ON names.gid = name_with_duplicates.gid AND names.ntype = name_with_duplicates.ntype,
+                    (SELECT @runninggroup := 0) AS b,
+                    (SELECT @previousgid := 0) AS prevGid,
+                    (SELECT @previousntype := 0) AS previousNType
+                    ORDER  BY names.gid, names.ntype, name_with_duplicates.ftype) T;
+
+            -- Create new udflds records for names with incremented number that don't yet exist
+            INSERT INTO udflds (ftable, ftype, fcode, fname, ffmt, fdesc, lfldno, fuid, fdate, scaleid)
+            SELECT udflds.ftable, udflds.ftype, toCreate.fcode_with_increment, udflds.fname, udflds.ffmt, udflds.fdesc, udflds.lfldno,
+                udflds.fuid, udflds.fdate, udflds.scaleid
+                FROM (SELECT DISTINCT name_with_incremented_number.ntype, name_with_incremented_number.fcode_with_increment
+                    FROM name_with_incremented_number
+                    WHERE NOT EXISTS (SELECT 1
+                        FROM udflds
+                        WHERE name_with_incremented_number.fcode_with_increment = udflds.fcode
+                            AND udflds.ftable = 'NAMES')) toCreate
+                        INNER JOIN udflds ON toCreate.ntype = udflds.fldno;
+
+            -- Update the duplicated names with newly created names
+            UPDATE names n
+                INNER JOIN name_with_incremented_number ON n.nid = name_with_incremented_number.nid
+                LEFT JOIN udflds ON name_with_incremented_number.fcode_with_increment = udflds.fcode AND udflds.ftable = 'NAMES'
+                SET n.ntype = udflds.fldno;
```

Comment:
Add condition here: 

WHERE n.ntype <> udflds.fldno;

To make sure that only names that do not match the updated udflds.fldno will be updated.